segmentation_labeling_app/transforms/transformations.py

```diff
@@ -0,0 +1,212 @@
+from pathlib import Path
+import random
+from typing import Union, Tuple
+import os.path
+import math
+import copy
+
+import h5py
+import cv2
+import numpy as np
+import imageio_ffmpeg as mpg
+
+
+def downsample_h5_video(video_path: Union[str, Path], input_fps: int,
+                     output_fps: int, strategy: str = 'random') -> np.ndarray:
+    """
+    Function to down-sample an array stored in h5 format to a desired FPS.
+    Follows the strategy specified in the function call, current strategies are
+    (random, maximum, average). Random takes a random sample of one frame from
+    window, maximum takes the maximum frame in window, average takes the mean
+    of window.
+    Args:
+        video_path: The path to the input video in str or Path format
+        input_fps: The FPS or Hz of the input video
+        output_fps: The desired output FPS or Hz
+        strategy: The down-sampling startegy to follow
+
+    Returns:
+        new_video: A new video down-sampled to desired FPS with selected
+        strategy
+    """
+    if os.path.exists(video_path):
```

Comment:
just curious, earlier you were a strong proponent of pathlib, I thought, and here you are using both `os` and `pathlib` when most things could be pathlib:
https://docs.python.org/3/library/pathlib.html#correspondence-to-tools-in-the-os-module
any reason?

Reply:
It was to be agnostic to str vs path input, you can use os on path objects but not the other way around. I think for simplicity I'll remove the agnostic input type.